backend/tests/v5/unit/services/filesManager.test.js

```diff
@@ -189,7 +190,78 @@ const testGetFileAsStream = () => {
 	});
 };
 
+const testStoreFile = () => {
+	describe('Store file in fileshare', () => {
+		test('should throw error if the default type is not recognised', async () => {
+			const { defaultStorage } = config;
+			config.defaultStorage = 'unrecognised storage type';
+
+			await expect(FilesManager.storeFile(
+				generateRandomString(),
+				generateRandomString(),
+				generateRandomString(),
+			)).rejects.toEqual(templates.fileNotFound);
+
+			config.defaultStorage = defaultStorage;
+		});
+
+		test('should store file if default storage type is fs', async () => {
+			const { defaultStorage } = config;
+			config.defaultStorage = 'fs';
+
+			const refInfo = { _id: generateRandomString() };
+			FSHandler.storeFile.mockResolvedValueOnce(refInfo);
+			const data = generateRandomString();
+
+			await expect(FilesManager.storeFile(generateRandomString(),
+				generateRandomString(), data)).resolves.toEqual(refInfo);
+			expect(FSHandler.storeFile).toHaveBeenCalledTimes(1);
+			expect(FSHandler.storeFile).toHaveBeenCalledWith(data);
+
+			config.defaultStorage = defaultStorage;
+		});
+
+		test('should store file if default storage type is null and fs is set', async () => {
```

Comment:
I added that because I have this method and I need to cover all scenarios :

![image](https://user-images.githubusercontent.com/71635617/174297461-d7ec96e9-0438-45ca-a27a-ee4b3da9aaa8.png)

do you suggest this method should be placed in config ?

Reply:
kind of, but not this function particularly.

Config [sets a bunch of defaults for anything that isn't filled in](https://github.com/3drepo/3drepo.io/blob/master/backend/src/v4/config.js#L108), so we should add to it so ti determins the `defaultStorageType` if it wasn't configured.

so on the rest of the code base we can assume defaultStorage is always configured.